Automattic/jetpack
# Tiled gallery module: add srcset to images

## Proposed changes:

In an effort to move away from the `devicepx` script towards native functionality, this PR adds a `srcset` to images in the tiled gallery module (not the block). The browser will automatically pick the most appropriate image from the options in the `srcset`, depending on the screen pixel ratio and the current zoom value.

* Add `srcset` to the images in the tiled gallery module
* Add `loading=lazy` to the images in the tiled gallery module
* Reimplement grayscale functionality as a CSS filter, rather than as an additional image

Note: the grayscale filter will not work in IE and some other older browsers. Given that those browsers are no longer supported, that the grayscale feature appears to be deprecated, and that the failure mode is to display the images in color, this should easily qualify as graceful degradation.

### Other information:

- N/A Have you written new tests for your changes, if applicable?

## Jetpack product discussion
None.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

To create a test gallery on a page:
- Using the classic editor or a classic block in the block editor, click the "Add Media" button
- Pick several images and choose one of the gallery options, such as "Tiled Mosaic"
- Save the gallery and the page

To create a test gallery with the grayscale feature:
- Start with a regular gallery as described above
- Enter the code editor
- Add `grayscale="true"` to the gallery shortcode, e.g. `[gallery type="square" grayscale="true" ids="143,142,141,135"]`

To test the gallery:
- Open the post containing the gallery
- Inspect the images and ensure they include a `srcset` attribute
- Ensure they also include a `loading` attribute, set to `lazy`
- For grayscale galleries, ensure the images are initially displayed in grayscale, but are shown in color as you hover over them
